packages/ibc-utils/src/neutron.rs

```diff
@@ -38,19 +41,45 @@ pub fn ibc_send_message(
             .min_fee,
     );
     let total_fee = flatten_ntrn_ibc_fee(&ibc_fee);
-    let amount_minus_fee = amount
-        .checked_sub(total_fee.u128())
-        .ok_or_else(|| StdError::generic_err("Amount too low to pay for IBC transfer fees."))?;
+
+    // Sender's account balance for fee denom (NTRN)
+    let sender_ntrn_balance = deps.querier.query_balance(sender, NTRN_DENOM)?.amount;
+
+    let transfer_amount = if denom.to_string() == NTRN_DENOM {
+        if Uint128::from(amount) == sender_ntrn_balance {
+            // The full balance is being transferred .. deduct the fee from the transfer amount
+            Uint128::from(amount).checked_sub(total_fee)?
+        } else {
+            // Check that the balance is sufficient to cover the fees
+            let amount_plus_fee = total_fee.checked_add(amount.into())?;
+            if sender_ntrn_balance < amount_plus_fee {
+                return Err(StdError::generic_err(format!(
+                    "Insufficient balance to cover for IBC fees '{}' in sender account (required: {}, available: {}).",
+                    NTRN_DENOM, amount_plus_fee, sender_ntrn_balance,
+                )));
+            }
+            Uint128::from(amount)
+        }
+    } else {
+        if sender_ntrn_balance < total_fee {
+            return Err(StdError::generic_err(format!(
+                "Insufficient balance to cover for IBC fees '{}' in sender account (required: {}, available: {}).",
+                NTRN_DENOM, total_fee, sender_ntrn_balance,
+            )));
+        }
+        amount.into()
+    };
+
     let coin = Coin {
-        denom: denom.clone(),
-        amount: amount_minus_fee.to_string(),
+        denom: denom.to_string(),
+        amount: transfer_amount.to_string(),
     };
 
-    let msg = match denom_to_pfm_map.get(&denom) {
+    let msg = match denom_to_pfm_map.get(&denom.to_string()) {
         None => neutron_sdk::proto_types::neutron::transfer::MsgTransfer {
             source_port: "transfer".to_string(),
             source_channel: channel.clone(),
-            sender,
+            sender: sender.to_string(),
             receiver: to.clone(),
             token: Some(coin),
             timeout_height: Some(cosmos_sdk_proto::ibc::core::client::v1::Height {
```

Comment:
I know this is not the part of this PR but are we sure we want to use height-based timeouts? I don't think hardcoding revision number to 2 is right

Reply:
@bekauz hardcoded value was wrong, my bad. And I agree with your remark .. height-based timeouts are a bad idea to deal with, as they require some understanding about the underlying infrastructure (e.g. client revision number). Time-based timeouts provide a simpler (and more stable) abstraction. I have removed the ability to specify height-based timeouts, let me know what you think.